### PR TITLE
Initial work on setting host dynamically

### PR DESCRIPTION
**Pull request by itsAndL**

### CODE REVIEW

**Comment by itsAndL** [comment]
@mohamedrez, setting the ENV variables in ApplicationController it's not working

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,10 +3,16 @@
 class ApplicationController < ActionController::Base
   include Scopes
 
+  before_action :set_host_with_port
   before_action :set_paper_trail_whodunnit
   before_action :set_locale
   before_action :authenticate_user!
 
+  def set_host_with_port
+    ENV["HOST_WITH_PORT"] = request.host_with_port
```
Comment (mohamedrez) [inline app/controllers/application_controller.rb:12]: does not wrok ENV should not be set in the code

**Review by mohamedrez — CHANGES_REQUESTED** [review]

**Comment by itsAndL** [comment]
@mohamedrez i overload the environment variables in action mailer
but with Devise and routes.rb file, it's not working

**Comment by itsAndL** [comment]
[Change password is not working for akdital.humaneo.ma (the link root is wrong : uat.humaneo)](https://trello.com/c/87Q2NfeE/217-change-password-is-not-working-for-akditalhumaneoma-the-link-root-is-wrong-uathumaneo)

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -24,7 +24,7 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = "no-reply@uat.humaneo.ma"
+  config.mailer_sender = "no-reply@#{ENV["HOST_WITH_PORT"]}"
```
Comment (itsAndL) [inline config/initializers/devise.rb:27]: @mohamedrez  this one stay with the default value

File: config/environments/production.rb [hunk 3]
```diff
@@ -1,5 +1,8 @@
 require "active_support/core_ext/integer/time"
 
+ENV["HOST_WITH_PORT"] ||= "uat.humaneo.ma:8800"
```
Comment (mohamedrez) [inline config/environments/production.rb:3]: not needed we can add those inside the machine

**Review by mohamedrez — CHANGES_REQUESTED** [review]

**Comment by mohamedrez** [comment]
ref https://github.com/mohamedrez/grh/pull/477

**Pull request closed without merging**